pkg/collector/python/check.go

```diff
@@ -330,6 +338,58 @@ func (c *PythonCheck) ID() check.ID {
 	return c.id
 }
 
+// GetDiagnoses returns the diagnoses cached in last run or diagnose explicitly
+func (c *PythonCheck) GetDiagnoses() ([]diagnosis.Diagnosis, error) {
+	// Lock the GIL and release it at the end of the run (will crash otherwise)
+	gstate, err := newStickyLock()
+	if err != nil {
+		log.Warnf("failed to get lock for check %s: %s", c.id, err)
+		return nil, err
+	}
+	defer gstate.unlock()
+
+	// Get JSON serialized diagnoses. Handcrafted and significantly more complicated
+	// manual serialization was only 2-2.5 times faster and hence not worth it for
+	// low-rate calls like this
+	pyDiagnoses := C.get_check_diagnoses(rtloader, c.instance)
+	if pyDiagnoses == nil {
+		// When no actual diagnoses to report python check normally returns "[]"
+		log.Warnf("check diagnose failed to collect diagnoses JSON for %s", c.id)
+		return nil, nil
+	}
+	defer C.rtloader_free(rtloader, unsafe.Pointer(pyDiagnoses))
+
+	// Deserialize it
+	strDiagnoses := C.GoString(pyDiagnoses)
+
+	var diagnosesWrap []diagnosisJSONSerWrap
+	err = json.Unmarshal([]byte(strDiagnoses), &diagnosesWrap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse diagnoses JSON for %s: %s. JSON: %s", c.id, err, strDiagnoses)
```

Comment:
Would it make sense to use `%q` to quote any special characters that the json string may contain? This would avoid breaking log line with newlines, and also make it easier to spot any funny characters that may be causing parse errors in the first place.

```suggestion
		return nil, fmt.Errorf("failed to parse diagnoses JSON for %s: %s. JSON: %q", c.id, err, strDiagnoses)
```